mountd/worker.c

```diff
@@ -220,10 +220,14 @@ static int rpc_request(struct ksmbd_ipc_msg *msg)
 {
 	struct ksmbd_rpc_command *req;
 	struct ksmbd_rpc_command *resp;
-	struct ksmbd_ipc_msg *resp_msg;
+	struct ksmbd_ipc_msg *resp_msg = NULL;
 	int ret = -ENOTSUP;
 
+	if (msg->sz < sizeof(struct ksmbd_rpc_command))
+		goto out;
+
 	req = KSMBD_IPC_MSG_PAYLOAD(msg);
+	req->payload_sz = msg->sz - sizeof(struct ksmbd_rpc_command);
```

Comment:
Looking at this again, I don't think this line is necessary. Without it, I had crashes while fuzzing, but that's because my code was wrong to begin with (it only set req->sz, without touching req->payload_sz).